### DISCUSSION TITLE
Export CSV

### DISCUSSION THREAD

**Discussion by mzyc64** [solectrus/solectrus#2306] [Ideas]
Aus meinem Wechselrichter (SolarEdge) habe ich die Möglichkeit per CSV je Tag/Monat/Jahr erzeugten Strom in eine CSV zu Exportieren. 
Daraus habe ich mir in einem XLS eine Auswertung gebaut, notiere darin zusätzlich abgelesen Werte vom Stromzähler, und berechne mir einige Werte. 
Mit den Werten setze ich mich dann auseinander, vergleiche Zeiträume, vergleiche mit meiner Kalkulation vor Anschaffung, vergleiche mit meinen Vater und Freunden etc. 

Für einzelne Tage übernehme ich manuell aus der SENEC App die Tageswerte für 

- Erzeugung
- aus Netz bezogen
- in Netz eingespeist
- in Speicher
- aus Speicher

um diese zu vergleichen. Für größere Zeiträume ist es mir zu aufwändig, gerne hätte ich diese jedoch auch durchgehend. 
Im Grunde wäre dies die Top10 Liste (bzw. die Datengrundlage hierzu), nicht auf TOP Werte sondern nach Datum gefiltert. Diese nach CSV exportiert wäre das Pendant zum Export vom Wechselrichter. Wäre dies denkbar? 

Besteht darüber hinaus die Möglicheit zum Export der Datenbankdaten bzw. wäre eine Export-Funktion denkbar?  

Noch nutze ich Solectus nicht, da meine NAS dies nicht unterstützt. Einen RaspberryPi werde ich bestellen. Somit habe ich mich bisher nur mit der Demo-Version auf der Webseite auseinander gesetzt. Viele Schritte in meinem XLS könnte ich mir mit SOLECTUS und einer dauerhaften Speicherung von Daten sparen, tolle Arbeit! 

Grüße 
Christian

**Comment by ledermann** [comment]
Einen Export von Daten aus der SOLECTRUS-Oberfläche habe ich aus heutiger Sicht nicht vor. Da aber die Messwerte allesamt in InfluxDB liegen, kann man die Export-Funktion des dortigen Admin-Frontends nutzen. Somit kommt man an alles dran. Auch konkrete Auswertungen lassen sich (mit entsprechenden Kenntnissen) in InfluxDB erstellen und dann exportieren.

Vieles von dem, was du oben aufgeführt hast, ist in SOLECTRUS direkt enthalten und läuft automatisiert ab. Insofern sparst du dir viel Zeit und Arbeit, wenn du statt Excel zukünftig SOLECTRUS verwendest. 

Es wäre übrigens denkbar, die CSV-Daten von SolarEdge zu importieren, sodass alle historischen Daten seit Installation der PV-Anlage verfügbar sind. Für SENEC und Sungrow funktioniert das bereits, [für Kostal ist es geplant](https://github.com/solectrus/csv-importer/issues/69).

**Comment by q2doc** [comment]
zwei Fragen - ich ärgere mich gerade mit InfluxDB 2 rum, der CSV Export Knopf ist weg und die CLI liefert nicht das was ich will. Da ich aber Solectrus auf den Pi5 umziehen will, die Frage wie ich am einfachsten die Daten mitnehme?
Und dann kam der Blick auf den CSV-Importer, mein SMA Sunny Tripower bietet ja auch den Export im 5-Minuten-Raster an... ich stelle mal eine CSV in die dortige Diskussion 😉

**Reply by ledermann** [reply]
Zum Export fallen mir noch zwei Möglichkeiten ein:

1) InfluxDB-Container herunterfahren. Datenordner kopieren. Am neuen Ort einfügen. Hatte das schon mal erfolgreich gemacht. Geht auch mit der PostgreSQL-Datenbank.
2) Außerdem gibt es ja noch das offizielle Backup-Tool: https://docs.influxdata.com/influxdb/v2/admin/backup-restore/backup/
